infrastructure/SharedInfrastructure/src/SharedInfrastructure/CoreStack.cs

```diff
@@ -0,0 +1,269 @@
+                {
+                    { "UserPoolId", WebAppUserPool.UserPoolId },
+                    { "Username", UserName },
+                    { "TemporaryPassword", "P@ssword1" },
+                    { "MessageAction", "SUPPRESS" }
+                },
+                PhysicalResourceId = PhysicalResourceId.Of($"{Constants.AppName}AdminUser")
+            },
+            OnDelete = new AwsSdkCall
+            {
+                Service = "CognitoIdentityServiceProvider",
+                Action = "adminDeleteUser",
+                Parameters = new Dictionary<string, string>
+                {
+                    { "UserPoolId", WebAppUserPool.UserPoolId },
+                    { "Username", UserName }
+                }
+            },
+            Policy = AwsCustomResourcePolicy.FromSdkCalls(new SdkCallsPolicyOptions { Resources = AwsCustomResourcePolicy.ANY_RESOURCE })
+        });
+
+        var adminUserAttachment = new CfnUserPoolUserToGroupAttachment(this, "AttachAdminUserToAdministratorsGroup", new CfnUserPoolUserToGroupAttachmentProps
+        {
+            GroupName = CognitoAdminUserGroup.GroupName,
+            Username = UserName,
+            UserPoolId = WebAppUserPool.UserPoolId
+        });
+
+        adminUserAttachment.Node.AddDependency(defaultUser);
+    }
+
+    internal void CreateUserPoolClient()
+    {
+        var localClient = new UserPoolClient(this, "LocalClient", new UserPoolClientProps
+        {
+            UserPool = WebAppUserPool,
+            GenerateSecret = false,
+            PreventUserExistenceErrors = true,
+            ReadAttributes = new ClientAttributes()
+                    .WithStandardAttributes(new StandardAttributesMask
+                    {
+                        GivenName = true,
+                        FamilyName = true,
+                        Email = true
+                    }),
+            SupportedIdentityProviders = new[]
+            {
+                UserPoolClientIdentityProvider.COGNITO
+            },
+            AuthFlows = new AuthFlow
+            {
+                UserPassword = true
+            },
+            OAuth = new OAuthSettings
+            {
+                Flows = new OAuthFlows
+                {
+                    AuthorizationCodeGrant = true
+                },
+                Scopes = new[]
+                {
+                    OAuthScope.OPENID,
+                    OAuthScope.EMAIL,
+                    OAuthScope.COGNITO_ADMIN,
+                    OAuthScope.PROFILE
+                },
+                CallbackUrls = new[]
+                {
+                    $"{userPoolCallbackUrlRoot}/signin-oidc"
+                },
+                LogoutUrls = new[]
+                {
+                    $"{userPoolCallbackUrlRoot}/"
+                }
+            }
+        });
+
+        var adminSiteUserPoolDomain = WebAppUserPool.AddDomain($"{Constants.AppName}UserPoolDomain", new UserPoolDomainOptions
```

Comment:
do we need the admin prefix here on the variable name, now that we have only one pool? I got confused and went looking for an admin pool on first read.